File: apps/tv/modules/valence-edge-fade/ios/ValenceEdgeFadeModule.swift
import ExpoModulesCore

/// Offers JavaScript a view that fades what it holds out towards one of its edges.
public class ValenceEdgeFadeModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ValenceEdgeFade")

    View(ValenceEdgeFadeView.self) {
      Prop("edge") { (view: ValenceEdgeFadeView, edge: String) in
        view.edge = edge
      }

      Prop("reach") { (view: ValenceEdgeFadeView, reach: Double) in
        view.reach = CGFloat(reach)
      }
    }
  }
}

File: apps/tv/modules/valence-edge-fade/android/src/main/java/app/valence/tv/edgefade/ValenceEdgeFadeModule.kt
package app.valence.tv.edgefade

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/** Offers JavaScript a view that fades what it holds out to nothing towards one edge. */
class ValenceEdgeFadeModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ValenceEdgeFade")

    View(ValenceEdgeFadeView::class) {
      Prop("edge") { view: ValenceEdgeFadeView, edge: String -> view.edge = edge }
      Prop("reach") { view: ValenceEdgeFadeView, reach: Float -> view.reach = reach }
    }
  }
}
